engine/class_modules/sc_death_knight.cpp

```diff
@@ -5503,6 +5522,18 @@ struct pillar_of_frost_bonus_buff_t : public buff_t
   }
 };
 
+struct pillar_of_frost_buff_rank_2_t : public buff_t
+{
+  pillar_of_frost_buff_rank_2_t(death_knight_t* p) :
+    buff_t(p, "pillar_of_frost_rank_2")
+  {
+    set_duration( p -> spec.pillar_of_frost -> duration() );
+    set_default_value( p -> spec.pillar_of_frost_rank_2 -> effectN( 1 ).percent() );
+
+    add_invalidate( CACHE_STRENGTH );
+  }
+};
```

Comment:
Another case where `apply_affecting_aura` seemed to not work.  Doing it this way also offers the benefit of explicitly listing the rank 2 in the buffs section of the output, which makes it clear to the user that rank 2 is present as the base Pillar of Frost has the rank 1 tooltip.

Reply:
According to spelldata, pillar of frost rank 2 only increases the amount of strength on the buff.
You shouldn't need a buff for that, just a spelldata object and a modifier on the default value of pillar of frost's buff.
I don't think adding yet another pillar of frost buff helps visibility, especially when the user can check the html report for pillar of frost's buff value, which should be either 0.10 or 0.20, depending on spell ranks.

apply_affecting_aura only works for whitelisted effects, it is irrelevant for pillar of frost since it only increases your Strength.

Reply:
Oh that would be a lot more straight forward.  To do that I would just modify the call to `set_default_value` in the pillar of frost buff right to also add the rank 2 value?